python/GafferUI/SpreadsheetUI.py

```diff
@@ -1212,6 +1315,276 @@ def __fixedWidth( cls, plugValueWidget ) :
+					Gaffer.Metadata.registerValue( self.__rowsPlug, "spreadsheet:section:{}:index".format( sectionName ), i )
+
+	def __columnAdded( self, cellsPlug, cellPlug ) :
+
+		self.__updateTabs()
+
+	def __columnRemoved( self, cellsPlug, cellPlug ) :
+
+		self.__updateTabs()
+
+	def __renameSection( self, sectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+		assert( sectionName in sectionNames )
+		sectionIsCurrent = self._qtWidget().tabText( self._qtWidget().currentIndex() ) == sectionName
+
+		newSectionName = GafferUI.TextInputDialogue(
+			title = "Rename section",
+			initialText = sectionName,
+			confirmLabel = "Rename",
+		).waitForText( parentWindow = self.ancestor( GafferUI.Window ) )
+
+		if not newSectionName or newSectionName == sectionName :
+			return
+
+		if newSectionName in sectionNames :
+			for suffix in range( 1, len( sectionNames ) + 2 ) :
+				suffixedSectionName = newSectionName + str( suffix )
+				if suffixedSectionName not in sectionNames :
+					newSectionName = suffixedSectionName
+					break
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+
+			# Move appropriate columns to renamed section
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					Gaffer.Metadata.registerValue( cellPlug, "spreadsheet:section", newSectionName )
+
+			# Reapply section order for renamed section.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ newSectionName if n == sectionName else n for n in sectionNames ]
+			)
+
+			# And choose the renamed tab if necessary.
+			if sectionIsCurrent :
+				self._qtWidget().setCurrentIndex( sectionNames.index( sectionName ) )
+
+	def __deleteSection( self, sectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+			# Iterate in reverse to avoid invalidating column indices we're
+			# about to visit.
+			for columnIndex, cellPlug in reversed( list( enumerate( self.__rowsPlug["default"]["cells"] ) ) ) :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					self.__rowsPlug.removeColumn( columnIndex )
+			# Reassign section indices to remove gap.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ n for n in sectionNames if n != sectionName ]
+			)
+
+	def __moveSection( self, fromSectionName, toSectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+		sectionIsCurrent = self._qtWidget().tabText( self._qtWidget().currentIndex() ) == fromSectionName
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+
+			# Move columns
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
```

Comment:
Would it be worth wrapping this up into a helper method? There are a bunch of places where we do the 'do I have metadata if not then remap `""` to `Other`' dance that could easily get out of sync (just thinking of the catalogue fun weve had of late that was down to a similar duplication of logic).

Reply:
Tidied this all up in d35774d, which uses setSection()/getSection() instead of direct metadata access.